### PR TITLE
products/shipping/discount list settings

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

**PR intended to be tested with API branch:** feature/multichannel-multicurrency

### Screenshots
Screenshot 2020-09-30 at 13 13 09

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1113 Products/sales/vouchers/shipping list settings](https://app.clickup.com/t/2549495/SALEOR-1113)

**Review by orzechdev — CHANGES_REQUESTED** [review]
When I don't have any channel created, I'm still getting dialog opened when opening the product list page:
Zrzut ekranu 2020-09-30 o 17 35 23

File: src/discounts/views/SaleList/SaleList.tsx [hunk 1]
```diff
@@ -156,6 +156,9 @@ export const SaleList: React.FC<SaleListProps> = ({ params }) => {
 
   const handleSort = createSortHandler(navigate, saleListUrl, params);
   const currencySymbol = maybe(() => shop.defaultCurrency, "USD");
+  const onSettingsOpen = !!channelChoices.length
+    ? { onSettingsOpen: () => openModal("settings") }
+    : {};
```
Comment (orzechdev) [inline src/discounts/views/SaleList/SaleList.tsx:161]: I think it is easier to read:
```suggestion
  const onSettingsOpen = !!channelChoices.length
    ? () => openModal("settings")
    : undefined;
```

File: src/discounts/views/SaleList/SaleList.tsx [hunk 2]
```diff
@@ -219,7 +224,7 @@ export const SaleList: React.FC<SaleListProps> = ({ params }) => {
                 </IconButton>
               }
               selectedChannel={selectedChannel}
-              onSettingsOpen={() => openModal("settings")}
+              {...onSettingsOpen}
```
Comment (orzechdev) [inline src/discounts/views/SaleList/SaleList.tsx:227]: Or even:
```suggestion
              onSettingsOpen={!!channelChoices.length ?
                  () => openModal("settings")} :
                  undefined
```

**Review by orzechdev — APPROVED** [review]

**Pull request merged**